devito/data/data.py

```diff
@@ -365,3 +560,151 @@ def index_handle_oob(idx):
                              "multidimensional index arrays")
     else:
         return idx
+
+
+def local_data_idx(loc_idx):
+    """Retrieve the unflipped idx corresponding to loc_idx."""
+    loc_data_idx = []
+    for i in as_tuple(loc_idx):
+        if isinstance(i, slice) and i.step is not None and i.step == -1:
+            if i.stop is None:
+                loc_data_idx.append(slice(0, i.start+1, -i.step))
+            else:
+                loc_data_idx.append(slice(i.stop+1, i.start+1, -i.step))
+        elif isinstance(i, slice) and i.step is not None and i.step < -1:
+            if i.stop is None:
+                lmin = i.start
+                while lmin >= 0:
+                    lmin += i.step
+                loc_data_idx.append(slice(lmin-i.step, i.start+1, -i.step))
+            else:
+                loc_data_idx.append(slice(i.stop+1, i.start+1, -i.step))
+        elif is_integer(i):
+            loc_data_idx.append(slice(i, i+1, 1))
+        else:
+            loc_data_idx.append(i)
+    return as_tuple(loc_data_idx)
+
+
+class SLICING(Tag):
+    pass
+MPI = SLICING('mpi')  # noqa
```

Comment:
Can we rename that, we already have MPI in devito so maybe confusing.

like distributed_slice and serial_slice or something

Reply:
How about `PARALLEL_SLICE` and `SERIAL_SLICE` or just `PARALLEL`/`SERIAL`?